core/storage/suggestion/gae_models.py

```diff
@@ -386,3 +386,92 @@ def update_position_in_rotation(cls, score_category, user_id):
         else:
             instance.current_position_in_rotation = user_id
             instance.put()
+
+
+class GeneralVoiceoverApplicationModel(base_models.BaseModel):
+    """A general model for voiceover application of an entity.
+
+    The ID of the voiceover application will be a random hashed value.
+    """
+    # The type of entity to which the user will be assigned as a voice artist
+    # once the application will get approved.
+    target_type = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the entity to which the application belongs.
+    target_id = ndb.StringProperty(required=True, indexed=True)
+    # The language code for the voiceover audio.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The status of the application. One of: accepted, rejected, in-review.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # The HTML content written in the given language_code.
+    # This will typically be a snapshot of the content of the initial card of
+    # the target.
+    content = ndb.StringProperty(required=True, indexed=True)
```

Comment:
Should this actually be indexed? Should it be TextProperty? What's the length limit?

Reply:
Done!